core/src/main/scala/io/snappydata/SnappyStoreTableSizeProviders.scala

```diff
@@ -0,0 +1,188 @@
+  @volatile
+  private var tableSizeInfo = Map[String, Long]()
+  private var timer: SystemTimer = null
+
+
+  def start(sc: SparkContext): Unit = {
+    val delay =
+      sc.getConf.getOption("spark.snappy.calcTableSizeInterval")
+          .getOrElse(DEFAULT_CALC_TABLE_SIZE_SERVICE_INTERVAL).toString.toLong
+      Misc.getGemFireCache.getCCPTimer().schedule(calculateTableSizeTask(sc), delay, delay)
+  }
+
+  def calculateTableSizeTask(sc: SparkContext): SystemTimer.SystemTimerTask = {
+    new SystemTimer.SystemTimerTask {
+      override def run2(): Unit = {
+        val currentTableSizeInfo = mutable.HashMap[String, Long]()
+
+        val result =
+          FunctionService.onMembers(GfxdMessage.getAllGfxdServers())
+              .withCollector(new GfxdListResultCollector())
+              .execute(RegionSizeCalculatorFunction.ID).getResult()
+
+        result.asInstanceOf[ArrayList[HashMap[String, Long]]]
+            .asScala.foreach(_.asScala.foreach(row => {
+          var totalSize: Long = row._2
+          if (currentTableSizeInfo.contains(row._1)) {
+            totalSize = currentTableSizeInfo.get(row._1).get + row._2
+          }
+          currentTableSizeInfo.put(row._1, totalSize)
+        }))
+
+          tableSizeInfo = currentTableSizeInfo.toMap
+      }
+
+      override def getLoggerI18n: LogWriterI18n = {
+        return Misc.getGemFireCache.getLoggerI18n
+      }
+    }
+  }
+
+  def getTableSize(tableName: String, isColumnTable: Boolean = false):
+  Option[Long] = {
+    val currentTableSizeInfo = tableSizeInfo
+    if (currentTableSizeInfo == null || !currentTableSizeInfo.contains(tableName)) {
+      None
+    }
+    else {
+      if (isColumnTable) {
+        val optStat = currentTableSizeInfo.get(ColumnFormatRelation.cachedBatchTableName(tableName))
+        if (optStat.isDefined) {
+          Some(optStat.get + currentTableSizeInfo.get(tableName).get)
+        } else {
+          None
+        }
+      } else {
+        Some(currentTableSizeInfo.get(tableName).get)
+      }
+    }
+  }
+
+}
+
+object StoreTableSizeProvider {
+
+  def getTableSizes: Seq[UIAnalytics] = {
+    val currentTableStats = tryExecute(conn => getMemoryAnalyticsDetails(conn))
+    if (currentTableStats == null) {
+      return Seq.empty
+    }
+    val internalColumnTables = currentTableStats.filter(entry => isColumnTable(entry._1))
+
+    var columnTableNames = scala.collection.mutable.Seq[String]()
+
+    (internalColumnTables.map(entry => {
+      val rowBuffer = getRowBufferName(entry._1)
+      val rowBufferSize = currentTableStats.get(rowBuffer).get
+      columnTableNames = columnTableNames.+:(rowBuffer).+:(entry._1)
+      new UIAnalytics(rowBuffer, rowBufferSize, entry._2, true)
+    })
+        ++ currentTableStats.filter(
+      entry => {
+        !columnTableNames.contains(entry._1)
+      }).map(rowEntry =>
+      new UIAnalytics(rowEntry._1, rowEntry._2, 0, false))).toSeq
+  }
```

Comment:
Can't you do in one pass instead of double filter like:
```scala
currentTableStats.map {
  case (isColumnTable(k), v) =>
      val rowBuffer = getRowBufferName(entry._1)
      val rowBufferSize = currentTableStats.get(rowBuffer).get 
      new UIAnalytics(rowBuffer, rowBufferSize, v, true)
  case (k, v) => new UIAnalytics(k, v, 0, false)
}
```